samples/firefox-accounts/src/main/java/org/mozilla/samples/fxa/MainActivity.kt

```diff
@@ -96,10 +99,35 @@ open class MainActivity : AppCompatActivity() {
                     txtView.text = getString(R.string.signed_in, "${value.displayName ?: ""} ${value.email}")
                 }
                 account?.toJSONString().let {
-                    getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE).edit().putString(FXA_STATE_KEY, it).apply()
+                    getSharedPreferences(FXA_STATE_PREFS_KEY, Context.MODE_PRIVATE).edit()
+                            .putString(FXA_STATE_KEY, it).apply()
                 }
             }
             account?.completeOAuthFlow(code, state)?.then(handleAuth)?.whenComplete(handleProfile)
         }
     }
+
+    private fun openWebView(url: String) {
+        supportFragmentManager?.beginTransaction()?.apply {
+            replace(R.id.container, LoginFragment.create(url, REDIRECT_URL))
+            addToBackStack("open webView")
+            commit()
+        }
+    }
+
+    override fun onLoginComplete(code: String, state: String, fragment: LoginFragment) {
+        val txtView: TextView = findViewById(R.id.txtView)
+        val handleAuth = { _: OAuthInfo -> account?.getProfile() }
+        val handleProfile = { value: Profile ->
+            runOnUiThread {
+                txtView.text = getString(R.string.signed_in, "${value.displayName ?: ""} ${value.email}")
+            }
+        }
+
+        account?.completeOAuthFlow(code, state)?.then(handleAuth)?.whenComplete(handleProfile)
+        supportFragmentManager?.beginTransaction()?.apply {
+            remove(fragment)
+            commit()
+        }
```

Comment:
Above you added the fragment to the backstack. I assume you just want to pop the backstack here instead of manually removing the fragment?
Mixing backstack and non-backstack transactions can have weird unexpected results where suddenly fragments come back when pressing back. :)

Reply:
[backstack'd](https://github.com/mozilla-mobile/android-components/pull/438/commits/776c5e9ff7a40cccd5dc29e68d5cbc2fee9508fb#diff-1cb1220d920ea31717d5e4089234fc4dR128)